### ISSUE TITLE
Learning to Solve NP-Complete Problems - A Graph Neural Network for the Decision TSP

### ISSUE THREAD

**Issue by QiXuanWang** [QiXuanWang/LearningFromTheBest#34]
Link: [SemanticScholar](https://www.semanticscholar.org/paper/Learning-to-Solve-NP-Complete-Problems-A-Graph-for-Prates-Avelar/960d80dfff5cddf7ad16edcc95027ac9ddec2166)

Comment:
Published in AAAI 2018
Looks like a naive try of GNN on TSP. One point that worth studying is "Now that we have obtained a solver for the decision TSP, we can exploit it to yield route cost predictions within a reasonable margin from the optimal cost."
The author also achieved "1024 instances with 80% accuracy" and "loses accuracy when size goes up"

Problem:
Combinational Optimization

Contribution:
GNNs employ trainable modules which can be assembled in different configurations that reflect the relational structure of each problem instance. In this paper, we show that GNNs can learn to solve, with very little supervision, the decision variant of the Traveling Salesperson Problem (TSP), a highly relevant NP-Complete problem. Our model is trained to function as an effective message-passing algorithm in which edges (embedded with their weights) communicate with vertices for a number of iterations after which the model is asked to decide whether a route with cost < C exists. 
...
In summary, our work shows that Graph Neural Networks are powerful enough to solve NP-Complete problems which combine symbolic and numeric data.
